packages/dd-trace/src/gateway/engine/runner.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+const als = require('../als')
+const lockName = 'ddRunnerLock' // TODO: do we really need this (to be global) ?
+
+function runSubscriptions (subscriptions, params) {
+  const results = []
+
+  if (!subscriptions.size || process[lockName]) return results
+  process[lockName] = true
+
+  const store = als.getStore()
+
+  const executedCallbacks = new Set()
+
+  subscriptions.forEach((subscription) => {
```

Comment:
Interesting. Do you have data for that? also comparing against `Set#values()`.

If we're really in a hot path here, it may also be worth comparing against using an Array instead of a Set. A microbenchmark around this code might be helpful.

Reply:
Ok nevermind foreach is trash, even without closures and with various sizes it's slower than iterator. I've tried other stuff too and `for of` is the fastest for Sets. I have no clue why I thought forEach was faster as I've always thought it was slower.
I'll change it